### PR TITLE
Feat/gpu/signed eq ne

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: backends/tfhe-cuda-backend/cuda/include/integer.h [hunk 1]
```diff
@@ -140,7 +140,7 @@ void scratch_cuda_integer_radix_comparison_kb_64(
     uint32_t pbs_level, uint32_t pbs_base_log, uint32_t grouping_factor,
     uint32_t lwe_ciphertext_count, uint32_t message_modulus,
     uint32_t carry_modulus, PBS_TYPE pbs_type, COMPARISON_TYPE op_type,
-    bool allocate_gpu_memory);
+    bool is_signed, bool allocate_gpu_memory);
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply backends/tfhe-cuda-backend/cuda/include/integer.h:143]: Yeah I guess it was just easier :thinking:

File: tfhe/src/integer/gpu/server_key/radix/tests_unsigned/test_comparison.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+/// - unchecked_$comparison_name
+/// - $comparison_name
+///
+/// So, for example, for the `gt` comparison fn, this macro will generate the tests for
+/// the 2 variants described above
+macro_rules! define_gpu_comparison_test_functions {
+    ($comparison_name:ident, $clear_type:ty) => {
+        ::paste::paste!{
+            fn [<integer_unchecked_ $comparison_name _ $clear_type:lower>]<P>(param: P) where P: Into<PBSParameters>{
+                let num_tests = 1;
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply tfhe/src/integer/gpu/server_key/radix/tests_unsigned/test_comparison.rs:22]: Yes indeed, I haven't changed it though as I think this should be the topic of another PR dedicated to tests probably

File: tfhe/src/integer/server_key/radix_parallel/tests_signed/test_comparison.rs [hunk 3]
```diff
@@ -0,0 +1,821 @@
...
+    // Some hard coded tests
+    let pairs = [
+        (Scalar::ONE, Scalar::TWO),
+        (Scalar::TWO, Scalar::ONE),
+        (-Scalar::ONE, Scalar::ONE),
+        (Scalar::ONE, -Scalar::ONE),
+        (-Scalar::ONE, -Scalar::TWO),
+        (-Scalar::TWO, -Scalar::ONE),
+        (Scalar::ZERO, -Scalar::ONE),
+        (Scalar::MAX, Scalar::ZERO),
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed/test_comparison.rs:61]: Yes, I just took them as they were :slightly_smiling_face:

File: tfhe/src/integer/server_key/radix_parallel/tests_signed/test_comparison.rs [hunk 4]
```diff
@@ -0,0 +1,821 @@
...
+            create_parametrized_test!([<integer_signed_unchecked_ $comparison_name _ $clear_type>]
+            {
+
+                PARAM_MESSAGE_2_CARRY_2_KS_PBS,
+
+                PARAM_MESSAGE_3_CARRY_3_KS_PBS,
+
+                PARAM_MESSAGE_4_CARRY_4_KS_PBS,
+                #[cfg(tarpaulin)]
+                COVERAGE_PARAM_MESSAGE_2_CARRY_2_KS_PBS
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed/test_comparison.rs:358]: No it was not, I found it strange as well

File: tfhe/src/integer/gpu/server_key/radix/comparison.rs [hunk 5]
```diff
@@ -507,12 +542,12 @@ impl CudaServerKey {
                 (ct_left, &tmp_rhs)
             }
             (false, true) => {
-                tmp_lhs = ct_right.duplicate_async(stream);
+                tmp_lhs = ct_left.duplicate_async(stream);
                 self.full_propagate_assign_async(&mut tmp_lhs, stream);
                 (&tmp_lhs, ct_right)
             }
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply tfhe/src/integer/gpu/server_key/radix/comparison.rs:548]: Yes all good :+1:

File: tfhe/src/integer/gpu/server_key/radix/tests_unsigned/test_comparison.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/gpu/server_key/radix/tests_unsigned/test_comparison.rs:22]: 💯

File: tfhe/src/integer/gpu/ciphertext/boolean_value.rs [hunk 6]
```diff
@@ -20,20 +21,23 @@ use crate::shortint::Ciphertext;
 /// Also, some function such as comparisons are known to return an encrypted value
 /// that is either 0 or 1, and thus return a CudaCiphertext wrapped in a [CudaBooleanBlock].
 pub struct CudaBooleanBlock {
-    pub ciphertext: CudaRadixCiphertext,
+    pub ciphertext: CudaUnsignedRadixCiphertext,
 }
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/gpu/ciphertext/boolean_value.rs]: ok !

File: tfhe/src/integer/server_key/radix_parallel/tests_signed/test_comparison.rs [hunk 7]
```diff
@@ -0,0 +1,821 @@
...
+    let pairs = [
+        (Scalar::ONE, Scalar::TWO),
+        (Scalar::TWO, Scalar::ONE),
+        (-Scalar::ONE, Scalar::ONE),
+        (Scalar::ONE, -Scalar::ONE),
+        (-Scalar::ONE, -Scalar::TWO),
+        (-Scalar::TWO, -Scalar::ONE),
+        (Scalar::ZERO, -Scalar::ONE),
+        (Scalar::MAX, Scalar::ZERO),
+    ];
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed/test_comparison.rs:461]: Follow up issue is here: https://github.com/zama-ai/tfhe-rs-internal/issues/517

[the remark this answers is not in the document]
Comment (agnesLeroy) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed/test_comparison.rs:461]: I put it in the tfhe-rs Kanban

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed/test_comparison.rs:461]: thank you very much 🙏

**Pull request merged**